pkg/kubectl/cmd/autoscale.go

```diff
@@ -118,62 +110,73 @@ func RunAutoscale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []
 		return cmdutil.UsageError(cmd, fmt.Sprintf("generator %q not found.", generatorName))
 	}
 	names := generator.ParamNames()
-	params := kubectl.MakeParams(cmd, names)
-	name := info.Name
-	params["default-name"] = name
 
-	params["scaleRef-kind"] = mapping.GroupVersionKind.Kind
-	params["scaleRef-name"] = name
-	params["scaleRef-apiVersion"] = mapping.GroupVersionKind.GroupVersion().String()
+	for i := range infos {
```

Comment:
Why not use Visit?

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/create.go#L121